src/messages.rs

```diff
@@ -68,6 +68,28 @@ impl From<Message> for Vec<u8> {
     }
 }
 
+impl TryFrom<libp2p::gossipsub::Message> for Message {
```

Comment:
You can just combine the previous identify_topic() and deserialize_message() into one. Implement TryFrom<(libp2p::gossipsub::TopicHash, libp2p::gossipsub::Message)>.

Just for reminder, I think there 2 problems in your current solution:
1. it's always weird to switch something back to string and do comparison. (which i understand in this case you couldn't thought of other ways, but the solution we used before should work)
2.  Its not a good idea to treating the rest of the cases as hotstuff_rs messages. I guess it wont cause error but it is not a good practice to do that in general.

Reply:
`libp2p::gossipsub::TopicHash` is essentially part of `libp2p::gossipsub::Message` as shown in the code block below. 

````
    pub struct Message {
    /// Id of the peer that published this message.
    pub source: Option<PeerId>,

    /// Content of the message.
    pub data: Vec<u8>,

    /// A random sequence number.
    pub sequence_number: Option<u64>,

    /// The topic this message belongs to
    pub topic: TopicHash,
    }
````
I think its redundant to TryFrom a tuple since TopicHash can be de-structured from Message itself. 

Previously, we used identify_topics(local_peer_address) to get Topics, and the variable local_peer_address is needed to successfully do this conversion. So I will instead implement this form of tuple instead: 

````
 impl TryFrom<(libp2p::gossipsub::Message, pchain_types::cryptography::PublicAddress)> for Message
````

Reply:
Oh right, I missed that part. That sounds good, go ahead.